src/editmap.cpp

```diff
@@ -686,7 +686,7 @@ void editmap::draw_main_ui_overlay()
         } else {
 #endif
             hilights["mapgentgt"].draw( *this, true );
-            tmpmap.reset_vehicle_cache( target.z );
+            tmpmap.rebuild_vehicle_cache( target.z );
```

Comment:
Since `rebuild_vehicle_cache` now doesn't call `clear_vehicle_cache`, you probably need to clear the cache explicitly before calling `rebuild_vehicle_cache`?